paintings/src/main/java/com/ataulm/artcollector/paintings/ui/PaintingsActivity.kt

```diff
@@ -24,14 +27,19 @@ class PaintingsActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_paintings)
 
-        val paintingsAdapter = PaintingsAdapter(picasso)
+        val paintingsAdapter = PaintingsAdapter(picasso) { viewModel.onClick(it) }
         recyclerView.apply {
             adapter = paintingsAdapter
             layoutManager = GridLayoutManager(this@PaintingsActivity, 2)
         }
 
-        viewModel.paintings.observe(this, Observer<List<Painting>> { paintings ->
-            paintings?.let { paintingsAdapter.submitList(it) }
+        viewModel.paintings.observe(this, DataObserver<List<Painting>> { paintings ->
+            paintingsAdapter.submitList(paintings)
+        })
+
+        viewModel.events.observe(this, EventObserver {
+            val painting = it.painting
+            startActivity(Navigation.PAINTING.viewIntent(painting.artist.id, painting.id))
```

Comment:
Navigation is in the base module, and `Painting` is defined twice, once in `paintings` and once in `painting`. They're technically different, so we just pass the IDs as strings.